### DISCUSSION TITLE
About the upcoming update

### DISCUSSION THREAD

**Discussion by Leroy8888** [HelixNebulaStudio/RiseOfTheDead#67] [General]
There still are a few things that I feel should be adressed before release. Will close this post after the release of the update.

Mending and the Nekrosis Amplifier aren't obtainable at the moment besides being purchasable from Icarus. Is it intentional that these 2 mods are locked behind a gold paywall?

Flinch is a mechanic that I feel like should be discussed a bit. As stat ''flinch protection'' could be important to some builds. Unlike tool condition, which doesn't really effect the game besides staring at a rusty or shiny weapon a randomly generated number of flinch protection on hats does. I've yet again created a poll about it, but by the looks it may take a bit to gather more votes on it. See the message in the unof discord server regarding it in the poll channel.

Is the concept about inferior mods included in this update or will it be kept for possible future content?

The codex also states the deagle can be bought from the RAT store for 200 perks, unsure if intentional, but it's not in the RAT store at the moment.

Tanker's rebar currently also cannot be obtained in any way.

Robert seems to be replaced by regular bosses when the corrosponding mission isn't active. While it does make sense in a logical matter it really does not seem to fit in a game where a players kills the same entity for example Mothena, multiple times for rewards.

**Comment by MXKhronos** [comment]
Thanks for bringing those up.

- I just added `mendingmod` and `nekrosampmod` to the `zenithcrate` and `veinofnekron` droptable respectively.
- I read the discussing in unof a little bit and made some changes to compromise.
    - The flinch idea is mainly for headwear items because it relates to the head. So I'm aware that some people won't enjoy the camera being flinched when getting attacked.
    - I've reduced the camera flinch strength by half and flinching will now contribute more to weapon inaccuracy.
    - I've increased `flinchprotection` mod's max protection to 75% from 50% just so it weighs a bit more than the rng flinch protection from obtaining the headwear. 
    - Also complaints about stuns but only the zomborg bosses can stun so I reduced the durations of the stuns a bit.
- Inferior mods will stay and we could discuss more about it since the overall max health of basic zombies are lowered a few commits back, I hope the balance isn't too far off.
    - The idea for augments would be a commonly obtained drop from zombies which will allow you to temporarily increase a mod's tier for a month, a bit of monthly weapon maintenance to super charge its capacity.
- Fixed `deagle`'s config in rat shop.
- `tankerrebar` won't be obtainable, it was just designed to be equipped by `Tanker`.
- Since the `Missing In Action` mission is replayable, I think it's fine for Robert to only show up when the mission is active.

**Comment by Leroy8888** [comment]
Alright, thank you for clarifying those things. However there still are a few things I would like to discuss or point out.

You stated that you increased the protection of a max flinch mod, but also stated that you will keep the RNG stat from headwear. The poll that was created seems to have most people favor hatwear gaining a flat value over the RNG as a lot of people disliked the idea of having RNG stats on headwear. The poll also was mainly created to adjust the flinch addition on hatwear in a way that it is acceptable or enjoyable by a majority of the community and as such, was there to have people vote for one. It wasn't exactly created to be the deciding factor if the mechanic should be added or removed.

But despite that, it seems that the change for RNG headwear was kept when majority of the poll favored the outcome of set values for this status on all headwears.

The poll might be small of sample size for that matter, but I unfortunately cannot gather more as I can only reach to a part of the community, which should be kept in mind. Though, seeing the outcome it is likely that regardless of size, the outcome would stay the same.


The monthly upkeep might also be a point of discussion. As it introduces another task to the daily grind, which a lot of people won't like. These daily tasks usually hinder players in achieving the goals they actually want to achieve, as they have to go through the dailies / occasional tasks first. It might also result in people simple avoiding high tier weapons all together, making possible future content such as stronger weapons not relevant for people.

It can of course be tested and reviewed to see the overall response of the mechanic first upon launch, though gathering some opinions from the community could potentially avoid a lot of fuss with having people spend time, tweak and adjust their weapons overall to be changed yet again a few days later.

**Comment by MXKhronos** [comment]
The rng stat for headwear is mostly just for novelty and collectors, since headwear currently have a vase different set of utility, uniqueness in each headwear would give the same item reason to be compared. E.g. You enjoy wearing a `hardhat`, but it's 3% flich protection, then you find a 8% flinch protection `hardhat`. It probably won't make too much of a difference once you have a flinch mod on it but there's still a reason to pick one over the other. It's not something designed to be grinded for either, unless a player considers it fun to do so.

The upkeep nature for augments is ideally designed to reward playing, I guess if it's commonly dropped from zombies it would be grinded for which is not my intended design. I intended it not to be grinded for, perhaps a daily drop of sorts. It's also being designed to balance out hopefully balance out some power creep. If you've noticed, most of the newer weapons are designed to be more interesting to use rather than statistically stronger ideally.

**Comment by Leroy8888** [comment]
Alright. If the augments are a choice for balancing and the RNG aspect is not meant to be the go to protection, with the mod compensating for most of the power so be it.

Again thanks for the clarification on some of the choices for the upcoming changes

I'll leave the discussion open for if anyone feels like adding anything to it. I'll close this once the update is live.

**Comment by MrPigZ** [comment]
I like how low tier weapons are more customizable with more mods. Zombie hp nerf seems to work as intended. 

Here is a list of mod changes that should be made at very least (in my opinion):

- make focus mod high tier (without it tactical bow feels annoying to use and chaging it doesn't break the balance)
- Bow ricochet mod to high tier (reduces the amount of monthly maintenance)
- make explosion radius and explosive ammo cap mod to tier 2 (if these were to stay at tier 1 it would be a nerf to gl 😭)
- slugs mod to high tier (just give the 1 pierce to rusty 🥺 even though slugs don't go well with crits)
- match rocketman mod's tier to at4's (It feels weird that a mod which is made only for 1 weapon is inferior on it)
- smg crit multiplier to higher tier (gives the vector the uniqueness) 

I don't use pistols so I can't really say anything about them (make them somewhat 
low-maintenance weapons by increasing ammo cap, magazine and happy trigger mod tier? 🤔🤫🧏‍♂️) 

Bow mods most likely need some changes so that bow could be good at something cause currently what I tested rec beats both of them at everything.

**Comment by MXKhronos** [comment]
Made a bunch of adjustments to mod tiers, https://github.com/HelixNebulaStudio/RiseOfTheDead/commit/94cb8eddd1df8c32dd876d74079a87a2d40c9479.
Let me know if I missed anything.

**Comment by Leroy8888** [comment]
I am aware that I bring this up rather late, but I had a busy week and the upcoming unfortunately won't be any less stressfull. But eitherway, when I talked about the new update and the new content it could introduce someone brought up the weaponsmith.

This NPC is actually very anticipated by quite some people. It's unlikely that it will be an addition in this update, but when will it actually be planned to be released? The concept about it dates back quite a while.

I do think It would actually fit well within the anniversary update, especially now that we have these new weapon animations and a new mod tier system to boot. I could however see that the time window to get this addition in the game would be rather small.

**Reply by MXKhronos** [reply]
Likely next update along with augments. This update is quite packed and I'm starting to burn out so I want to push the update once it's ready and fix any remaining issues and perhaps take a very short break from RotD development.

**Comment by Leroy8888** [comment]
I see, uh, I did have something to mention about flinch cushioning yet again however.

The mod is locked behind genesis which is fairly late in the game, maybe it could appear somewhere a bit earlier? It might be rather difficult to deal with flinch on some occassions like bandit outpost otherwise.
